src/wpublish/QueriesAndChunks.cc

```diff
@@ -489,105 +499,168 @@ QueriesAndChunks::ScanTableSumsMap QueriesAndChunks::_calcScanTableSums() {
+            LOGS(_log, LOG_LVL_ERROR, __func__ << " Couldn't locate qIdToCheck=" << qIdToCheck);
+            // This really should never happen, but try to boot the next query in the list.
+            continue;
+        }
 
-/// @return true if this query is done and has not been touched for deadTime.
-bool QueryStatistics::isDead(chrono::seconds deadTime, chrono::system_clock::time_point now) {
-    lock_guard<mutex> guard(_qStatsMtx);
-    if (_isMostlyDead()) {
-        if (now - _touched > deadTime) {
-            return true;
+        if (!queryToCheck->_queryBooted) {
+            auto tasksBooted = queryToCheck->getTasksBooted();
+            LOGS(_log, LOG_LVL_DEBUG,
+                 __func__ << " b check uq=" << countQId.qId << " tasksBooted=" << tasksBooted
+                          << " max=" << (_maxTasksBooted));
+            if (tasksBooted > _maxTasksBooted) {
+                QueryStatistics::Ptr queryToBoot = getStats(qIdToCheck);
+                if (_bootUserQuery(queryToBoot, bSched)) {
+                    uqBootedTasksSum += countQId.count;
+                    LOGS(_log, LOG_LVL_DEBUG,
+                         __func__ << " b booted uq=" << countQId.qId << ":count=" << countQId.count
+                                  << " tasksBooted=" << tasksBooted
+                                  << " uqBootedTasksSum=" << uqBootedTasksSum);
+                }
+            }
         }
     }
-    return false;
+    return;
 }
 
-/// @return true if all Tasks for this query are complete.
-/// Precondition, _qStatsMtx must be locked.
-bool QueryStatistics::_isMostlyDead() const { return _tasksCompleted >= _size; }
+bool QueriesAndChunks::_bootUserQuery(QueryStatistics::Ptr queryToBoot,
+                                      wsched::BlendScheduler::Ptr const& bSched) {
+    // Get query info
+    if (queryToBoot == nullptr) {
+        LOGS(_log, LOG_LVL_ERROR, __func__ << " queryToBoot is nullptr");
+        // This really should never happen.
+        return false;
+    }
 
-ostream& operator<<(ostream& os, QueryStatistics const& q) {
-    lock_guard<mutex> gd(q._qStatsMtx);
-    os << QueryIdHelper::makeIdStr(q.queryId) << " time=" << q._totalTimeMinutes << " size=" << q._size
-       << " tasksCompleted=" << q._tasksCompleted << " tasksRunning=" << q._tasksRunning
-       << " tasksBooted=" << q._tasksBooted;
-    return os;
-}
+    // If the UserQuery was already booted, just add its booted `Task`s to the sum.
+    if (queryToBoot->_queryBooted) {
+        return true;
+    }
 
-void QueryStatistics::addTaskTransmit(double timeSeconds, int64_t bytesTransmitted, int64_t rowsTransmitted,
-                                      double bufferFillSecs) {
-    _histTimeTransmittingPerTask->addEntry(timeSeconds);
-    _histRowsPerTask->addEntry(rowsTransmitted);
-    _histSizePerTask->addEntry(bytesTransmitted);
-    _histTimeBufferFillPerTask->addEntry(bufferFillSecs);
-}
+    bool queryWasBooted = false;
```

Comment:
The logic flow above is a bit hard to read. Why not to do something like this instead:
```
bool queryWasBooted = queryToBoot->_queryBooted;
if (queryWasBooted) return true;
```